### PR TITLE
Home work 1.

### PR DESCRIPTION
**Pull request by interist32**
All tasks are distributed across separate commits.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,4285 @@
...
+      "optional": true,
+      "requires": {
+        "arr-flatten": "^1.0.1"
+      }
+    },
+    "arr-flatten": {
+      "version": "1.1.0",
+      "resolved": "https://registry.npmjs.org/arr-flatten/-/arr-flatten-1.1.0.tgz",
+      "integrity": "sha512-L3hKV5R/p5o81R7O02IGnwpDmkp6E982XhtbuwSe3O4qOtMMMtodicASA1Cny2U+aCXcNpml+m4dPsvsJ3jatg==",
+      "dev": true
```
Comment (MurDaD) [inline package-lock.json]: What version of npm are you using? You need to reinstall the packages and commit this file

File: hw-1/task1.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  input: process.stdin,
+  output: process.stdout,
+});
+
+rl.on('line', (text) => {
+  console.log(`${text} -> ${reverse(text)}`);
+});
+
+
+/**
```
Comment (MurDaD) [inline hw-1/task1.js]: Comments are mostly redundant in node js - you just need to name your methods to represent what they are doing. But it's ok, you can keep adding comments if you want

File: hw-1/task1.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import * as readline from 'readline';
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+rl.on('line', (text) => {
```
Comment (MurDaD) [inline hw-1/task1.js]: `rl` variable name is too short. It can be not so obvious what this variable for in big projects

File: hw-1/output.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{"book":"The Compound Effect","author":"Darren Hardy","amount":"5","price":"9.48"}
```
Comment (MurDaD) [inline hw-1/output.txt]: You need to commit this file with no content

File: hw-1/task2.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+const CSV_FILE_PATH = path.join(__dirname, './input.csv');
+const TEXT_FILE_PATH = path.join(__dirname, './output.txt');
+
+const readStream = require('fs').createReadStream(CSV_FILE_PATH);
+const writeStream = require('fs').createWriteStream(TEXT_FILE_PATH);
+
+readStream
+    .pipe(csv({
+            headers: ['book', 'author', 'amount', 'price'],
```
Comment (MurDaD) [inline hw-1/task2.js]: this is not universal solution, you need to think about solution that can handle different files

File: hw-1/task1.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  console.log(`${text} -> ${reverse(text)}`);
```
Comment (MurDaD) [inline hw-1/task1.js]: Minor: If the input is tabulation (TAB sign, but actually 6 spaces), then the output will be 2 spaces. Try to think, how to fix that. Example:
```
"       " -> "  "
```

File: package.json [hunk 7]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "node.js-mentoring-program",
+  "version": "1.0.0",
+  "description": "<img src=\"https://jsremotely.com/static/icons/nodejs.png\" width=\"100px\"/>",
+  "main": "index.js",
+  "scripts": {
+    "task1": "nodemon --exec babel-node ./hw-1/task1.js -I",
+    "task2": "nodemon --exec babel-node ./hw-1/task2.js",
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (MurDaD) [inline package.json]: There should be a 3rd task with babel

File: hw-1/task2.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+console.log(csv.csv);
+
+const CSV_FILE_PATH = path.join(__dirname, './input.csv');
+const TEXT_FILE_PATH = path.join(__dirname, './output.txt');
+
+const readStream = require('fs').createReadStream(CSV_FILE_PATH);
+const writeStream = require('fs').createWriteStream(TEXT_FILE_PATH);
+
+readStream
+    .pipe(csv({
```
Comment (MurDaD) [inline hw-1/task2.js]: You need to use 2 spaces, or 4 spaces everywhere, not different amount in different files

File: package-lock.json (hunk 1 above)
Comment (interist32) [reply package-lock.json]: Done. I use npm 6.13.2.

File: hw-1/task1.js (hunk 2 above)
Comment (interist32) [reply hw-1/task1.js]: Gotcha. Removed it.

File: hw-1/task1.js (hunk 3 above)
Comment (interist32) [reply hw-1/task1.js]: Fixed.

File: hw-1/output.txt (hunk 4 above)
Comment (interist32) [reply hw-1/output.txt]: Done

File: hw-1/task1.js (hunk 6 above)
Comment (interist32) [reply hw-1/task1.js]: Interesting. Do you have any ideas how to fix it? I noticed it happens if I use string concatenation. Though, I changed output style and it works, but I'm still interested what causes it?

File: package.json (hunk 7 above)
Comment (interist32) [reply package.json]: I was doing it task by task. I thought that after step 3 both programs have to use babel. As they do finally. Here's the list of commits: https://github.com/interist32/node.js-mentoring-program/pull/1/commits

File: hw-1/task2.js (hunk 5 above)
Comment (interist32) [reply hw-1/task2.js]: Done.

File: hw-1/task2.js (hunk 8 above)
Comment (interist32) [reply hw-1/task2.js]: Done

File: package.json (hunk 7 above)
Comment (MurDaD) [reply package.json]: ok, got it. can be accepted, but it was expected to run 2nd module with babel in seperate run script

File: hw-1/task1.js (hunk 6 above)
Comment (MurDaD) [reply hw-1/task1.js]: You need to define `UTF-8` encoding for input

Comment (interist32) [reply hw-1/task1.js]: Done! Thanks.

**Pull request merged**